src/components/Text/Text.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+import { stylesheet, style } from 'typestyle';
+import { ThemeConsumer, Theme } from '../Theme/Theme';
+/**
+ * Text component to style your `Text` effortlessly.
+ *
+ * **Usage**
+ * ```jsx
+ * <Text>Hello</Text>
+ * ```
+ */
+export const Text: React.FunctionComponent<TextProps> = (props) => {
+  const display = (theme: Theme) => {
+    let fontWeight: number = 400;
+    let fontStyle = 'normal';
+    const fontFamily = props.fontFamily;
+    const fontSize = `${props.fontSize}px`;
+    const color = props.color;
+    let padding: string;
+    let margin: string;
+    let textDecoration: string;
+    if (props.padding) {
+      if (Array.isArray(props.padding)) {
+        padding = props.padding.map((e) => `${e}px`).join(' ');
+      } else {
+        padding = `${props.padding}px`;
+      }
+    }
+    if (props.margin) {
+      if (Array.isArray(props.margin)) {
+        margin = props.margin.map((e) => `${e}px`).join(' ');
+      } else {
+        margin = `${props.margin}px`;
+      }
+    }
+    if (props.underline && props.strikeThrough) {
+      textDecoration = 'underline line-through';
+    } else {
+      if (props.underline) {
+        textDecoration = 'underline';
+      }
+      if (props.strikeThrough) {
+        textDecoration = 'line-through';
+      }
+    }
+    if (props.bold) {
+      fontWeight = 700;
+    } else {
+      fontWeight = 400;
+    }
+    if (props.italicize) {
+      fontStyle = 'italic';
+    } else {
+      fontStyle = 'normal';
+    }
+    const css: React.CSSProperties = {
+      fontWeight,
+      fontStyle,
+      fontSize,
+      fontFamily,
+      padding,
+      color,
+      margin,
+      textDecoration,
+    };
+    if (props.variant === 'h1') {
+      return <h1 style={css}>{props.children}</h1>;
+    }
+    if (props.variant === 'h2') {
+      return <h2 style={css}>{props.children}</h2>;
+    }
+    if (props.variant === 'h3') {
+      return <h3 style={css}>{props.children}</h3>;
+    }
+    if (props.variant === 'h4') {
+      return <h4 style={css}>{props.children}</h4>;
+    }
+    if (props.variant === 'h5') {
+      return <h5 style={css}>{props.children}</h5>;
+    }
+    if (props.variant === 'h6') {
+      return <h6 style={css}>{props.children}</h6>;
+    }
+    return <span style={css}>{props.children}</span>;
+  };
+  return (
+    <ThemeConsumer>
+      {(theme) => (
+        <p style={{display: props.inline ? 'inline' : 'block'}}>
```

Comment:
don't use `P` tag here.. this should be handled by `variant`.
also avoid using style. style should be available through props for consumer to customized it.